<div class="alert alert-block alert-info" style="margin-top: 20px">
    <a href="https://cocl.us/corsera_da0101en_notebook_top">
         <img src="https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/DA0101EN/Images/TopAd.png" width="750" align="center">
    </a>
</div>

<h1 align=center><font size=5>Data Science and Machine Learning Capstone Project</font></h1>
<h2 align=center><font size=3; color=blue> Copyright &#169; Mustafizur Rahman</font></h2>

<p><b>In this worksheet I will prepare Complaints Data for top 5 complaints type of all Boroughs</b></p>

<h3>Import data</h3>


In [1]:
import pandas as pd
import matplotlib.pylab as plt

<h2>Reading the data set from Computer storage folder.</h2>

In [2]:
filename = 'E:\\1.IPCOS BACKUP\\3. MMR_PSNL\\2020 Data Science and Analytics Courses\\3. Python Data Science Certificate\\5. Data Science and Machine Learning Capstone Project\\311_Service_Requests_from_2010_to_Present_min.csv'

Use the Pandas method <b>read_csv()</b> to load the data from local drive.

In [3]:
#df = pd.read_csv(filename, names = headers)
df = pd.read_csv(filename)
print(df.shape)
#df.size
df.tail(10)

(6019843, 16)


Unnamed: 0  Unique Key            Created Date  \
6019833     6019833    44063661  10/15/2019 10:13:11 PM   
6019834     6019834    44063670  10/15/2019 08:25:25 AM   
6019835     6019835    44063671  10/15/2019 09:32:54 AM   
6019836     6019836    44063680  10/15/2019 06:43:56 PM   
6019837     6019837    44063681  10/15/2019 11:35:19 AM   
6019838     6019838    44063685  10/15/2019 10:35:45 AM   
6019839     6019839    44063692  10/15/2019 06:09:28 AM   
6019840     6019840    44063724  10/15/2019 10:54:45 AM   
6019841     6019841    44063726  10/15/2019 02:07:34 PM   
6019842     6019842    44063737  10/15/2019 11:29:01 AM   

                    Closed Date        Complaint Type         Location Type  \
6019833  10/16/2019 02:07:21 AM        HEAT/HOT WATER  RESIDENTIAL BUILDING   
6019834  10/16/2019 09:55:24 PM        HEAT/HOT WATER  RESIDENTIAL BUILDING   
6019835  10/16/2019 07:02:48 PM        HEAT/HOT WATER  RESIDENTIAL BUILDING   
6019836  10/16/2019 05:55:02 PM        HEAT/HOT WATER  RESIDENTIAL BUILDING   
6019837                     NaN  UNSANITARY CONDITION  RESIDENTIAL BUILDING   
6019838  10/16/2019 05:55:02 PM        HEAT/HOT WATER  RESIDENTIAL BUILDING   
6019839  10/15/2019 04:12:58 PM        HEAT/HOT WATER  RESIDENTIAL BUILDING   
6019840  10/16/2019 06:14:29 PM        HEAT/HOT WATER  RESIDENTIAL BUILDING   
6019841  10/16/2019 07:10:19 PM        HEAT/HOT WATER  RESIDENTIAL BUILDING   
6019842  10/15/2019 09:19:45 PM         PAINT/PLASTER  RESIDENTIAL BUILDING   

         Incident Zip         Incident Address         Street Name  \
6019833       10452.0    957 WOODYCREST AVENUE   WOODYCREST AVENUE   
6019834       10470.0     707 EAST  242 STREET    EAST  242 STREET   
6019835       10458.0   2786 BAINBRIDGE AVENUE   BAINBRIDGE AVENUE   
6019836       10029.0     327 EAST  108 STREET    EAST  108 STREET   
6019837       10472.0    1630 EAST  174 STREET    EAST  174 STREET   
6019838       10029.0     327 EAST  108 STREET    EAST  108 STREET   
6019839       10461.0  3555 BRUCKNER BOULEVARD  BRUCKNER BOULEVARD   
6019840       10034.0       165 SHERMAN AVENUE      SHERMAN AVENUE   
6019841       10467.0    66 WEST GUN HILL ROAD  WEST GUN HILL ROAD   
6019842       10009.0             143 AVENUE D            AVENUE D   

        Address Type      City  Status  \
6019833      ADDRESS     BRONX  Closed   
6019834      ADDRESS     BRONX  Closed   
6019835      ADDRESS     BRONX  Closed   
6019836      ADDRESS  NEW YORK  Closed   
6019837      ADDRESS     BRONX    Open   
6019838      ADDRESS  NEW YORK  Closed   
6019839      ADDRESS     BRONX  Closed   
6019840      ADDRESS  NEW YORK  Closed   
6019841      ADDRESS     BRONX  Closed   
6019842      ADDRESS  NEW YORK  Closed   

                                    Resolution Description    Borough  \
6019833  The complaint you filed is a duplicate of a co...      BRONX   
6019834  The Department of Housing Preservation and Dev...      BRONX   
6019835  The Department of Housing Preservation and Dev...      BRONX   
6019836  The complaint you filed is a duplicate of a co...  MANHATTAN   
6019837  The following complaint conditions are still o...      BRONX   
6019838  The complaint you filed is a duplicate of a co...  MANHATTAN   
6019839  The Department of Housing Preservation and Dev...      BRONX   
6019840  The complaint you filed is a duplicate of a co...  MANHATTAN   
6019841  The Department of Housing Preservation and Dev...      BRONX   
6019842  The Department of Housing Preservation and Dev...  MANHATTAN   

          Latitude  Longitude  
6019833  40.831527 -73.928863  
6019834  40.905645 -73.849026  
6019835  40.868017 -73.888941  
6019836  40.791359 -73.940184  
6019837  40.834029 -73.875432  
6019838  40.791359 -73.940184  
6019839  40.847809 -73.827481  
6019840  40.864764 -73.922764  
6019841  40.883207 -73.884279  
6019842  40.724749 -73.975269

<b>Drop the Unnamed:0 column!</b>

In [4]:
df.drop('Unnamed: 0', axis=1, inplace=True)
df.head()

Unique Key            Created Date Closed Date        Complaint Type  \
0    45531130  02/02/2020 06:09:17 AM         NaN        HEAT/HOT WATER   
1    45529784  02/02/2020 02:15:24 PM         NaN  UNSANITARY CONDITION   
2    45527528  02/02/2020 02:27:41 AM         NaN        HEAT/HOT WATER   
3    45530329  02/02/2020 12:13:18 PM         NaN        HEAT/HOT WATER   
4    45528814  02/02/2020 01:59:44 PM         NaN             APPLIANCE   

          Location Type  Incident Zip         Incident Address  \
0  RESIDENTIAL BUILDING       10019.0     426 WEST   52 STREET   
1  RESIDENTIAL BUILDING       11204.0           1751 67 STREET   
2  RESIDENTIAL BUILDING       11372.0          87-15 37 AVENUE   
3  RESIDENTIAL BUILDING       10458.0  2405 SOUTHERN BOULEVARD   
4  RESIDENTIAL BUILDING       11209.0            223 78 STREET   

          Street Name Address Type             City Status  \
0    WEST   52 STREET      ADDRESS         NEW YORK   Open   
1           67 STREET      ADDRESS         BROOKLYN   Open   
2           37 AVENUE      ADDRESS  Jackson Heights   Open   
3  SOUTHERN BOULEVARD      ADDRESS            BRONX   Open   
4           78 STREET      ADDRESS         BROOKLYN   Open   

                              Resolution Description    Borough   Latitude  \
0  The following complaint conditions are still o...  MANHATTAN  40.765132   
1  The following complaint conditions are still o...   BROOKLYN  40.618484   
2  The following complaint conditions are still o...     QUEENS  40.750269   
3  The following complaint conditions are still o...      BRONX  40.853773   
4  The following complaint conditions are still o...   BROOKLYN  40.629745   

   Longitude  
0 -73.988993  
1 -73.992673  
2 -73.879432  
3 -73.881558  
4 -74.030533

 Check data Types and <b> Cardinality<b>. 

Identify_missing_values

<h4>Evaluating for Missing Data</h4>

The missing values are converted to Python's default. We use Python's built-in functions to identify these missing values. There are two methods to detect missing data:
<ol>
    <li><b>.isnull()</b></li>
    <li><b>.notnull()</b></li>
</ol>
The output is a boolean value indicating whether the value that is passed into the argument is in fact missing data.

In [5]:
missing_data = df.isnull()
missing_data.head(10)

Unique Key  Created Date  Closed Date  Complaint Type  Location Type  \
0       False         False         True           False          False   
1       False         False         True           False          False   
2       False         False         True           False          False   
3       False         False         True           False          False   
4       False         False         True           False          False   
5       False         False         True           False          False   
6       False         False         True           False          False   
7       False         False         True           False          False   
8       False         False         True           False          False   
9       False         False         True           False          False   

   Incident Zip  Incident Address  Street Name  Address Type   City  Status  \
0         False             False        False         False  False   False   
1         False             False        False         False  False   False   
2         False             False        False         False  False   False   
3         False             False        False         False  False   False   
4         False             False        False         False  False   False   
5         False             False        False         False  False   False   
6         False             False        False         False  False   False   
7         False             False        False         False  False   False   
8         False             False        False         False  False   False   
9         False             False        False         False  False   False   

   Resolution Description  Borough  Latitude  Longitude  
0                   False    False     False      False  
1                   False    False     False      False  
2                   False    False     False      False  
3                   False    False     False      False  
4                   False    False     False      False  
5                   False    False     False      False  
6                   False    False     False      False  
7                   False    False     False      False  
8                   False    False     False      False  
9                   False    False     False      False

<b>"True" stands for missing value</b>, while "False" stands for not missing value.

<h4>Count missing values in each column</h4>
<p>
Using a for loop in Python, we can quickly figure out the number of missing values in each column. As mentioned above, "True" represents a missing value, "False"  means the value is present in the dataset.  In the body of the for loop the method  ".value_counts()"  counts the number of "True" values. 
</p>

In [6]:
for column in missing_data.columns.values.tolist():
    print(column)
    print (missing_data[column].value_counts())
    print("") 
#for i in missing_data.columns.values.tolist():
    #print(i)
    #print (missing_data[i].value_counts())
    #print("")

Unique Key
False    6019843
Name: Unique Key, dtype: int64

Created Date
False    6019843
Name: Created Date, dtype: int64

Closed Date
False    5893186
True      126657
Name: Closed Date, dtype: int64

Complaint Type
False    6019843
Name: Complaint Type, dtype: int64

Location Type
False    5967019
True       52824
Name: Location Type, dtype: int64

Incident Zip
False    5939146
True       80697
Name: Incident Zip, dtype: int64

Incident Address
False    5967018
True       52825
Name: Incident Address, dtype: int64

Street Name
False    5967018
True       52825
Name: Street Name, dtype: int64

Address Type
False    5935078
True       84765
Name: Address Type, dtype: int64

City
False    5939569
True       80274
Name: City, dtype: int64

Status
False    6019843
Name: Status, dtype: int64

Resolution Description
False    6012017
True        7826
Name: Resolution Description, dtype: int64

Borough
False    6019843
Name: Borough, dtype: int64

Latitude
False    5939172
True       80671
N

Based on the summary above, each column has 6019843 rows of data, ten columns containing missing data:
<ol>
    <li>"Closed Date": 126657 missing data</li>
    <li>"Location Type": 52824 missing data</li>
    <li>"Incident Zip": 80697 missing data</li>
    <li>"Incident Address" : 52825 missing data</li>
    <li>"Street Name": 52825 missing data</li>
    <li>"Address Type": 84765 missing data</li>
    <li>"City": 80274 missing data</li>
    <li>"Resolution Description": 7826 missing data</li>
    <li>"Latitude": 80671 missing data</li>
    <li>"Longitude": 80671 missing data</li>

</ol>

<h3 id="deal_missing_values">Deal with missing data</h3>
<b>How to deal with missing data?</b>

<ol>
    <li>drop data<br>
        a. drop the whole row<br>
        b. drop the whole column
    </li>
    <li>replace data<br>
        a. replace it by mean<br>
        b. replace it by frequency<br>
        c. replace it based on other functions
    </li>
</ol>

We are not replacing any data in this case. As the data volume is huge we will drop whole rows with NaN values in "Incident Address" as this is the unique key we will use to merge PLUTO Data.

In [7]:
# simply drop whole row with NaN in "Incident Address" column
df.dropna(subset=["Incident Address"], axis=0, inplace=True)

# reset index, because we droped two rows
df.reset_index(drop=True, inplace=True)
df.head()

Unique Key            Created Date Closed Date        Complaint Type  \
0    45531130  02/02/2020 06:09:17 AM         NaN        HEAT/HOT WATER   
1    45529784  02/02/2020 02:15:24 PM         NaN  UNSANITARY CONDITION   
2    45527528  02/02/2020 02:27:41 AM         NaN        HEAT/HOT WATER   
3    45530329  02/02/2020 12:13:18 PM         NaN        HEAT/HOT WATER   
4    45528814  02/02/2020 01:59:44 PM         NaN             APPLIANCE   

          Location Type  Incident Zip         Incident Address  \
0  RESIDENTIAL BUILDING       10019.0     426 WEST   52 STREET   
1  RESIDENTIAL BUILDING       11204.0           1751 67 STREET   
2  RESIDENTIAL BUILDING       11372.0          87-15 37 AVENUE   
3  RESIDENTIAL BUILDING       10458.0  2405 SOUTHERN BOULEVARD   
4  RESIDENTIAL BUILDING       11209.0            223 78 STREET   

          Street Name Address Type             City Status  \
0    WEST   52 STREET      ADDRESS         NEW YORK   Open   
1           67 STREET      ADDRESS         BROOKLYN   Open   
2           37 AVENUE      ADDRESS  Jackson Heights   Open   
3  SOUTHERN BOULEVARD      ADDRESS            BRONX   Open   
4           78 STREET      ADDRESS         BROOKLYN   Open   

                              Resolution Description    Borough   Latitude  \
0  The following complaint conditions are still o...  MANHATTAN  40.765132   
1  The following complaint conditions are still o...   BROOKLYN  40.618484   
2  The following complaint conditions are still o...     QUEENS  40.750269   
3  The following complaint conditions are still o...      BRONX  40.853773   
4  The following complaint conditions are still o...   BROOKLYN  40.629745   

   Longitude  
0 -73.988993  
1 -73.992673  
2 -73.879432  
3 -73.881558  
4 -74.030533

<h3 id="correct_data_format">Correct data format</h3>
<b>We are almost there!</b>
<p>The last step in data cleaning is checking and making sure that all data is in the correct format (int, float, text or other).</p>

In Pandas, we use 
<p><b>.dtype()</b> to check the data type</p>
<p><b>.astype()</b> to change the data type</p>

<h4>Lets list the data types for each column</h4>

In [31]:
df.dtypes

Unique Key                  int64
Created Date               object
Closed Date                object
Complaint Type             object
Location Type              object
Incident Zip              float64
Incident Address           object
Street Name                object
Address Type               object
City                       object
Status                     object
Resolution Description     object
Borough                    object
Latitude                  float64
Longitude                 float64
dtype: object

<p>As we can see above, some columns are not of the correct data type. <b>Numerical variables should have type 'float' or 'int', and variables with strings should have type 'object'</b>. For example, 'Created Date' and 'Closed Date' variables are date time , so we should expect them to be of the type 'date-time' ; however, they are shown as type 'object'. We have to convert data types into a proper format for each column using the "astype()" method.</p> 

<h4>Convert data types to proper format</h4>
<p> By the way, in this problem we are not using Created Date and/or Closed Date for our predictive modeling. So I am not changing those dates here!</p>

In [21]:
#df[["Incident Zip"]] = df[["Incident Zip"]].astype("int") we cant change it integer as we have missing data "NaN"

In [23]:
#df.dtypes
df.describe(include=['object'])

Created Date             Closed Date  Complaint Type  \
count                  6019843                 5893186         6019843   
unique                 1675132                  757706              30   
top     01/24/2013 12:00:00 AM  11/07/2012 12:00:00 AM  HEAT/HOT WATER   
freq                      7581                    7296         1261574   

               Location Type Incident Address      Street Name Address Type  \
count                5967019          5967018          5967018      5935078   
unique                     5           182600             6825            1   
top     RESIDENTIAL BUILDING  34 ARDEN STREET  GRAND CONCOURSE      ADDRESS   
freq                 5959910            14298            92450      5935078   

            City   Status                             Resolution Description  \
count    5939569  6019843                                            6012017   
unique        89        5                                                340   
top     BROOKLYN   Closed  The Department of Housing Preservation and Dev...   
freq     2026580  5886253                                            1698990   

         Borough  
count    6019843  
unique         6  
top     BROOKLYN  
freq     1739886

<div class="alert alert-danger alertdanger" style="margin-top: 10px">
<h1> Question  #1: </h1>
<b>Which type of complaint should the Department of Housing Preservation and
Development of New York City focus on .</b>
</div>

In [8]:
df['Complaint Type'].value_counts()

HEAT/HOT WATER          1261574
HEATING                  887849
PLUMBING                 711130
GENERAL CONSTRUCTION     500863
UNSANITARY CONDITION     451643
PAINT - PLASTER          361257
PAINT/PLASTER            346438
ELECTRIC                 307310
NONCONST                 260890
DOOR/WINDOW              205278
WATER LEAK               193631
GENERAL                  151308
FLOORING/STAIRS          137402
APPLIANCE                112831
SAFETY                    51529
OUTSIDE BUILDING           7142
ELEVATOR                   6725
Unsanitary Condition       5499
CONSTRUCTION               5078
General                    1163
Safety                      424
STRUCTURAL                   16
Plumbing                     11
AGENCY                        9
Outside Building              6
VACANT APARTMENT              6
Appliance                     4
Electric                      1
Mold                          1
Name: Complaint Type, dtype: int64

<b>Wonderful!</b>

Now, we can see that we have 30 unique complaint type. Out of those "HEAT/HOT WATER" is maximum 1261574 complaint!. So the New York City should focus on <b>"HEAT/HOT WATER"</b> complaint type. Lets see how many other complaint type exist. 

<b>making the group based on top Complaint Type, boroughs,ZIP codes , and street. and Finally with top 5 Complaint Type with boroughs,ZIP codes , and street.

Hint: if you would like to select those columns use the following syntax: df[['Complaint Type','boroughs' ,'Incident Zip-Incident Address']].</b>


In [11]:
df_ComplType =df[['Complaint Type','Incident Zip','Incident Address','City','Borough']]
df_ComplType.head(20)
print(df_ComplType.size)
print(df_ComplType['Borough'].size)
df_ComplType['Borough'].value_counts()

29835090
5967018


BROOKLYN         1739886
BRONX            1617955
MANHATTAN        1055225
Unspecified       820397
QUEENS            645971
STATEN ISLAND      87584
Name: Borough, dtype: int64

In [35]:
is_heat_hot =  df_ComplType['Complaint Type']=='HEAT/HOT WATER'
df_ComplType_top1 =df_ComplType[is_heat_hot]#Data Frame with top complaint 'Heat/HOTWATER'
df_ComplType_top1.head(20)
print(df_ComplType_top1['City'].size)
df_ComplType_top1['City'].value_counts()

1261574


BRONX                  408970
BROOKLYN               382351
NEW YORK               283610
Elmhurst                17585
Jamaica                 17211
Flushing                15478
Astoria                 15324
STATEN ISLAND           12291
Woodside                 9161
Far Rockaway             9154
Ridgewood                8923
Corona                   8113
Jackson Heights          7782
Forest Hills             5911
Rego Park                4837
Sunnyside                4490
Kew Gardens              3792
Queens Village           2786
Long Island City         2664
South Richmond Hill      2635
Rockaway Park            2590
Richmond Hill            2493
Ozone Park               2464
South Ozone Park         2417
Hollis                   2271
East Elmhurst            2207
Saint Albans             2037
Springfield Gardens      1904
Woodhaven                1707
Arverne                  1533
Rosedale                 1488
Fresh Meadows            1283
Maspeth                  1179
Oakland Ga

In [36]:
print(df_ComplType_top1['Borough'].size)
df_ComplType_top1['Borough'].value_counts()

1261574


BRONX            410853
BROOKLYN         384523
MANHATTAN        285526
QUEENS           168318
STATEN ISLAND     12353
Unspecified           1
Name: Borough, dtype: int64

In [7]:
print(df_ComplType_top1['Incident Zip'].size)
df_Comp_zip=df_ComplType_top1['Incident Zip'].value_counts()
df_Comp_zip=df_Comp_zip.sort_values()
df_Comp_zip

1261574


12345.0        1
10103.0        1
10112.0        1
10129.0        1
10045.0        1
           ...  
10453.0    34241
10468.0    34507
10467.0    38110
10458.0    38864
11226.0    41786
Name: Incident Zip, Length: 191, dtype: int64

In [37]:
print(df_ComplType_top1['Incident Address'].size)
df_Comp_Adrr=df_ComplType_top1['Incident Address'].value_counts()
df_Comp_Adrr=df_Comp_Adrr.sort_values()
df_Comp_Adrr

1261574


958 EAST 101 STREET                1
128-09 NORTH CONDUIT AVENUE        1
589 DECATUR STREET                 1
1355 MYRTLE AVENUE                 1
142 WEST 123 STREET                1
                               ...  
2090 EAST TREMONT AVENUE        3615
9511 SHORE ROAD                 4587
34 ARDEN STREET                 4965
1025 BOYNTON AVENUE             5220
89-21 ELMHURST AVENUE          10896
Name: Incident Address, Length: 97979, dtype: int64

<b>Data Frame with 5 top complaint type.</b>


In [12]:
#Data Frame with top 5 complaint 'Heat/HOTWATER' and 'HEATING'
df_ComplType_top5=df_ComplType[(df_ComplType['Complaint Type']=='HEAT/HOT WATER') |(df_ComplType['Complaint Type']=='HEATING')|(df_ComplType['Complaint Type']=='PLUMBING')|(df_ComplType['Complaint Type']=='GENERAL CONSTRUCTION')|(df_ComplType['Complaint Type']=='UNSANITARY CONDITION')]
#df_ComplType_top5 =df_ComplType[top5_compl]#Data Frame with top 5 complaint type
print(df_ComplType_top5.shape)
df_ComplType_top5.head(20)


(3813059, 5)


Complaint Type  Incident Zip                   Incident Address  \
0         HEAT/HOT WATER       10019.0               426 WEST   52 STREET   
1   UNSANITARY CONDITION       11204.0                     1751 67 STREET   
2         HEAT/HOT WATER       11372.0                    87-15 37 AVENUE   
3         HEAT/HOT WATER       10458.0            2405 SOUTHERN BOULEVARD   
5         HEAT/HOT WATER       10456.0                 1131 MORRIS AVENUE   
7         HEAT/HOT WATER       11372.0                    37-20 81 STREET   
9         HEAT/HOT WATER       11207.0                   303 WYONA STREET   
10        HEAT/HOT WATER       10459.0                  1032 ALDUS STREET   
11        HEAT/HOT WATER       10039.0  2819 FREDERICK DOUGLASS BOULEVARD   
12              PLUMBING       11433.0           109-25 MERRICK BOULEVARD   
13  UNSANITARY CONDITION       11238.0                189 ST MARKS AVENUE   
14        HEAT/HOT WATER       11226.0                     297 LENOX ROAD   
15        HEAT/HOT WATER       10011.0               334 WEST   22 STREET   
16  UNSANITARY CONDITION       10456.0               488 EAST  164 STREET   
17        HEAT/HOT WATER       10026.0             225 CENTRAL PARK NORTH   
18        HEAT/HOT WATER       10031.0               529 WEST  151 STREET   
19        HEAT/HOT WATER       10473.0                 880 BOYNTON AVENUE   
20        HEAT/HOT WATER       11204.0                     2270 64 STREET   
21        HEAT/HOT WATER       11210.0               1655 FLATBUSH AVENUE   
22  UNSANITARY CONDITION       10457.0               1530 SHERIDAN AVENUE   

               City    Borough  
0          NEW YORK  MANHATTAN  
1          BROOKLYN   BROOKLYN  
2   Jackson Heights     QUEENS  
3             BRONX      BRONX  
5             BRONX      BRONX  
7   Jackson Heights     QUEENS  
9          BROOKLYN   BROOKLYN  
10            BRONX      BRONX  
11         NEW YORK  MANHATTAN  
12          Jamaica     QUEENS  
13         BROOKLYN   BROOKLYN  
14         BROOKLYN   BROOKLYN  
15         NEW YORK  MANHATTAN  
16            BRONX      BRONX  
17         NEW YORK  MANHATTAN  
18         NEW YORK  MANHATTAN  
19            BRONX      BRONX  
20         BROOKLYN   BROOKLYN  
21         BROOKLYN   BROOKLYN  
22            BRONX      BRONX

In [14]:
df_ComplType_top5['Complaint Type'].value_counts()

HEAT/HOT WATER          1261574
HEATING                  887849
PLUMBING                 711130
GENERAL CONSTRUCTION     500863
UNSANITARY CONDITION     451643
Name: Complaint Type, dtype: int64

In [43]:
print(df_ComplType_top5['Borough'].size)
df_ComplType_top5['Borough'].value_counts()

3813059


BROOKLYN         1066252
BRONX            1030715
MANHATTAN         680907
Unspecified       565490
QUEENS            419319
STATEN ISLAND      50376
Name: Borough, dtype: int64

In [15]:
df_ComplType_top5.to_csv("top5_compl_All_Borough.csv")

In [18]:
df_ComplType_top5_BRNX=df_ComplType_top5[(df_ComplType_top5['Borough']=='BRONX')]
df_ComplType_top5_BRNX.shape
#save the Bronx top 5 complain data in csv formate
df_ComplType_top5_BRNX.to_csv("top5_compl_BRONX.csv")

<b>Similarly creating and saving 5 top complaint type for other Borough.</b>


In [19]:
df_ComplType_top5_BROOKLN=df_ComplType_top5[(df_ComplType_top5['Borough']=='BROOKLYN')]
df_ComplType_top5_MNHTN=df_ComplType_top5[(df_ComplType_top5['Borough']=='MANHATTAN')]
df_ComplType_top5_QUEENS=df_ComplType_top5[(df_ComplType_top5['Borough']=='QUEENS')]
df_ComplType_top5_STISL=df_ComplType_top5[(df_ComplType_top5['Borough']=='STATEN ISLAND')]
#save the Bronx top 5 complain data in csv formate
df_ComplType_top5_BROOKLN.to_csv("top5_compl_BROOKLYN.csv")
df_ComplType_top5_MNHTN.to_csv("top5_compl_MNHTN.csv")
df_ComplType_top5_QUEENS.to_csv("top5_compl_QUEENS.csv")
df_ComplType_top5_STISL.to_csv("top5_compl_STISLND.csv")
print(df_ComplType_top5_BRNX.shape)
print(df_ComplType_top5_BROOKLN.shape)
print(df_ComplType_top5_MNHTN.shape)
print(df_ComplType_top5_QUEENS.shape)
print(df_ComplType_top5_STISL.shape)

(1030715, 5)
(1066252, 5)
(680907, 5)
(419319, 5)
(50376, 5)


<b>Data Frame with top 2 complaint type.</b>


In [20]:
#Data Frame with top 5 complaint 'Heat/HOTWATER' and 'HEATING'
df_ComplType_top2=df_ComplType_top5[(df_ComplType_top5['Complaint Type']=='HEAT/HOT WATER') |(df_ComplType_top5['Complaint Type']=='HEATING')]

print(df_ComplType_top2.shape)
df_ComplType_top2.head(20)

(2149423, 5)


Complaint Type  Incident Zip                   Incident Address  \
0   HEAT/HOT WATER       10019.0               426 WEST   52 STREET   
2   HEAT/HOT WATER       11372.0                    87-15 37 AVENUE   
3   HEAT/HOT WATER       10458.0            2405 SOUTHERN BOULEVARD   
5   HEAT/HOT WATER       10456.0                 1131 MORRIS AVENUE   
7   HEAT/HOT WATER       11372.0                    37-20 81 STREET   
9   HEAT/HOT WATER       11207.0                   303 WYONA STREET   
10  HEAT/HOT WATER       10459.0                  1032 ALDUS STREET   
11  HEAT/HOT WATER       10039.0  2819 FREDERICK DOUGLASS BOULEVARD   
14  HEAT/HOT WATER       11226.0                     297 LENOX ROAD   
15  HEAT/HOT WATER       10011.0               334 WEST   22 STREET   
17  HEAT/HOT WATER       10026.0             225 CENTRAL PARK NORTH   
18  HEAT/HOT WATER       10031.0               529 WEST  151 STREET   
19  HEAT/HOT WATER       10473.0                 880 BOYNTON AVENUE   
20  HEAT/HOT WATER       11204.0                     2270 64 STREET   
21  HEAT/HOT WATER       11210.0               1655 FLATBUSH AVENUE   
25  HEAT/HOT WATER       10034.0               502 WEST  213 STREET   
26  HEAT/HOT WATER       11238.0                 500 ST JOHNS PLACE   
28  HEAT/HOT WATER       11203.0               249 EAST   37 STREET   
29  HEAT/HOT WATER       10463.0           3110 KINGSBRIDGE TERRACE   
30  HEAT/HOT WATER       10462.0              2040 BRONXDALE AVENUE   

               City    Borough  
0          NEW YORK  MANHATTAN  
2   Jackson Heights     QUEENS  
3             BRONX      BRONX  
5             BRONX      BRONX  
7   Jackson Heights     QUEENS  
9          BROOKLYN   BROOKLYN  
10            BRONX      BRONX  
11         NEW YORK  MANHATTAN  
14         BROOKLYN   BROOKLYN  
15         NEW YORK  MANHATTAN  
17         NEW YORK  MANHATTAN  
18         NEW YORK  MANHATTAN  
19            BRONX      BRONX  
20         BROOKLYN   BROOKLYN  
21         BROOKLYN   BROOKLYN  
25         NEW YORK  MANHATTAN  
26         BROOKLYN   BROOKLYN  
28         BROOKLYN   BROOKLYN  
29            BRONX      BRONX  
30            BRONX      BRONX

<b>I'm not saving top 2 complaints Borough by borough! Can use all borough and filter it when needed.</b>


In [22]:
df_ComplType_top2.to_csv("top2_compl_All_Borough.csv")

<hr>
<p>Copyright &copy; 2020 Mustafizur Rahman <a href="https://www.linkedin.com/in/dr-mohammad-rahman-2434b313/">LinkedIn</a>.</p>
